### DISCUSSION TITLE
Is there any community search tool of homebrew to search casks and formulaes in third-party tap?

### DISCUSSION THREAD

**Discussion by IlIllIl** [Homebrew/discussions#4697] [Everyday usage]
### Output of `brew config`

```shell
-
```

### Output of `brew doctor`

```shell
-
```

### Description of issue

Hi, I'd like to know is there any command-line/web search tool for indexing and searching formula/cask in public third-party taps, similar to https://scoop.sh ? It's handy for me if I should find some isn't in official tap and manage them in one place.

**Comment by IlIllIl** [comment]
Why all of category need `brew config` and `brew doctor`? Sometimes it is unnecessary.

**Reply by osalbahr** [reply]
The questions are there because, more often than not, the answers are relevant even if OP doesn’t think so. And when in doubt, it doesn’t hurt to answer them.

For your case, though, yeah you can confidently know that it’s not that relevant. I’ve seen some people answer “N/A” and a “-“ is fine too. It’s somewhat relevant that you’re on macOS but you already have that away by mentioning casks.

**Comment by Bo98** [comment]
There is no central registry of third-party taps. You can host a tap anywhere without letting us know, but any search mechanism would require knowing a full list of available third-party taps.

**Comment by osalbahr** [comment]
Knowing how taps work, do you have suggestions for how a solution would look like? Would it be maintained by a third party? Maintained by Homebrew and is opt-in by the tap owner(s)?

As a workaround I recommend searching in https://formulae.brew.sh/analytics/install/365d/ (but it probably won’t show *all* formulae).

**Reply by apainintheneck** [reply]
That's a cool idea. I guess in theory it should be possible to use that to harvest 3rd/party tap names but we wouldn't be able to know where they are located for sure because they're just git repos. They could be anywhere.

Maybe it'd be better as an opt in sort of thing. If you'd like your tap to be included in this search, then just apply here. The other thing is that you'd probably have to update the search indexes regularly by cloning those taps and getting the list of formulae that way. If the third-party taps are small enough on average, which I assume they are, that might not be that bad but it's something to consider. You wouldn't want to ping for that info each time someone tries to search for something.

There's also the thing where we don't necessarily want to endorse the packages in other taps. It'd probably be necessary to add a note about that.

But at the end of the day I'm not sure there's much interest in something like this. Most people probably never use third-party taps to begin with since the core repo already has most packages people want to install.

**Reply by osalbahr** [reply]
Yeah, I agree that it’s a cool idea but like you said it’s not trivial to implement, especially with the endorsement security issue. Personally, I think the status quo is fine. It is what distro package managers do too. For example, I don’t think you can `apt`/`dnf` search globally in all third party repos.

The closest to this idea is the [AUR](https://wiki.archlinux.org/title/Arch_User_Repository), which is not just an index but something Arch Linux itself semi-monitors, which takes a lot of effort, and still not that secure (you’re still highly recommended to read the PKGBUILD diffs).

Side note: do you typically find a package through a repo *or* first find the package of interest then see the install options? Typically, when I want to install a CLI tool, I first try to see if it’s in Homebrew because that’s a “tap” that I trust. For lesser known packages, I typically go to the project’s README for instructions/options. You can from there ctrl+f for “brew” or “tap” to know if they support installing that way.

**Reply by apainintheneck** [reply]
I usually find the package first and then try to find out how I can install it. Sometimes it's not available from Homebrew so I end up using another package manager or just try to build from source.

**Reply by alebcay** [reply]
See also https://docs.brew.sh/Acceptable-Casks#homebrew-cask-is-not-a-discoverability-service. This is written specifically in the context of casks but I think it does apply for formulae etc. too.

**Comment by SMillerDev** [comment]
The closest thing is https://docs.brew.sh/Interesting-Taps-and-Forks#unsupported-interesting-taps that people can submit their tap to.

**Comment by IlIllIl** [comment]
Thanks for everyone, maybe creating a homebrew tap with command files is a good idea.
1. search for `homebrew-` in the GitHub Public Repository
2. make `brew tap-info someone/homebrew-tap --json` visualized
3. can search commands like casks and formulae
4. open homebrew tap homepage
5. if possible, autocreate a json index like official taps used?

As for security issues, I could review the repository content and source code to confirm that it is safe and usable.
